### PR TITLE
Refactor announcement

### PR DESCRIPTION
**Pull request by nekath**
- it adds hound config for eslint (and uses eslint only when checking javascript code)
- refactor announcement layout to react component

### CODE REVIEW

File: app/react/components/announcement-test.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import TestUtils from 'react-addons-test-utils';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+import expect from 'expect';
+
+import Announcement from './announcement';
+
+const props = {
+  propsCount: '9001',
```
Comment (zatapathique) [inline app/react/components/announcement-test.jsx]: It's over nine thousand.

File: app/react/components/announcement-test.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  it('renders', () => {
+    expect(TestUtils.findRenderedComponentWithType(component, Announcement)).toExist();
+  });
+
+  it('is animated', () => {
+    expect(TestUtils.findRenderedComponentWithType(component, ReactCSSTransitionGroup)).toExist();
+  });
+
+  it('fully shows after 500 miliseconds', () => {
+    const animation = TestUtils.findRenderedComponentWithType(component, ReactCSSTransitionGroup);
```
Comment (zatapathique) [inline app/react/components/announcement-test.jsx]: You could move this line to after the component has been rendered (line 14) and just reuse it in all of the tests. No need to be searching for the same component over and over.

File: app/react/components/announcement.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { animated: false };
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({animated: true});
```
Comment (zatapathique) [inline app/react/components/announcement.jsx]: Since you have spaces between hash bracers in other places, you should probably have them here as well :)

**Comment by zatapathique** [comment]
Everything looks great (pointed out some minor stuff you can consider to change).

File: app/react/components/announcement-test.jsx (hunk 2 above)
Comment (nekath) [inline app/react/components/announcement-test.jsx]: What do you mean exacly? Adding this to `before` callback would be an extra search for every test.

Comment (zatapathique) [inline app/react/components/announcement-test.jsx]: I don't mean to add this to before callback, you can add it to the main body of the test just like with the component rendering:
```
describe('announcement', () => {
    const component = TestUtils.renderIntoDocument(<Announcement {...props}/>);
    const animation = TestUtils.findRenderedComponentWithType(component, ReactCSSTransitionGroup);

# rest of the stuff
```

This way you render your component, you find the animation component and you just reuse it within the tests.

File: app/react/components/announcement.jsx (hunk 3 above)
Comment (nekath) [inline app/react/components/announcement.jsx]: somehow hound couldn't pick it, will fix

File: app/react/components/announcement-test.jsx (hunk 2 above)
Comment (nekath) [inline app/react/components/announcement-test.jsx]: ok, done

File: app/react/components/announcement-test.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+describe('announcement', () => {
+  const component = TestUtils.renderIntoDocument(<Announcement {...props}/>);
+  const animation = TestUtils.findRenderedComponentWithType(component, ReactCSSTransitionGroup);
+
+  it('renders', () => {
+    expect(TestUtils.findRenderedComponentWithType(component, Announcement)).toExist();
+  });
+
+  it('is animated', () => {
+    expect(TestUtils.findRenderedComponentWithType(component, ReactCSSTransitionGroup)).toExist();
```
Comment (zatapathique) [inline app/react/components/announcement-test.jsx]: You can also reuse it here `expect(animation).toExist();`

File: app/react/components/announcement-test.jsx (hunk 1 above)
Comment (lubieniebieski) [inline app/react/components/announcement-test.jsx]: :blue_heart:

**Pull request merged**